src/services/0_ExternToolsMS/hooks/index.js

```diff
@@ -0,0 +1,117 @@
+const { Forbidden, NotFound } = require('@feathersjs/errors');
+
+/**
+ * @param  {string, array[string]} inputPermissions
+ * @returns resolves if the current user has ANY of the given permissions
+ */
+const hasPermission = (inputPermissions) => {
+	const permissionNames = (typeof inputPermissions === 'string') ? [inputPermissions] : inputPermissions;
+
+	return (context) => {
+		const { params: { account, provider }, app } = context;
+		// If it was an internal call then skip this context
+		if (!provider) {
+			return Promise.resolve(context);
+		}
+
+		if (!account && !account.userId) {
+			throw new Forbidden('Can not read account data');
+		}
+
+		// Otherwise check for user permissions
+		return app.service('users').get(account.userId)
+			.then(({ permissions = [] }) => {
+				const hasAnyPermission = permissionNames.some((perm) => permissions.includes(perm));
+				if (!hasAnyPermission) {
+					throw new Forbidden(`You don't have one of the permissions: ${permissionNames.join(', ')}.`);
+				}
+				return Promise.resolve(context);
+			});
+	};
+};
+
+
+const testIfRoleNameExist = (user, roleNames) => {
+	if (typeof roleNames === 'string') {
+		roleNames = [roleNames];
```

Comment:
Codacy found an issue: [Assignment to function parameter 'roleNames'.](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-server/pullRequest?prid=6045798)